sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/FeatureManager.java

```diff
@@ -73,45 +96,174 @@ public Mono<Boolean> isEnabledAsync(String feature) {
+     * @return Assigned Variant
+     */
+    public Variant getVariant(String feature) {
+        return generateVariant(feature, null).block();
     }
 
-    private boolean checkFeature(String feature) throws FilterNotFoundException {
+    /**
+     * 
+     * @param feature Feature being checked.
+     * @param featureContext Local context
+     * @return Assigned Variant
+     */
+    public Variant getVariant(String feature, Object featureContext) {
+        return generateVariant(feature, featureContext).block();
+    }
+
+    /**
+     * 
+     * @param feature Feature being checked.
+     * @return Assigned Variant
+     */
+    public Mono<Variant> getVariantAsync(String feature) {
+        return generateVariant(feature, null).single();
+    }
+
+    /**
+     * 
+     * @param feature Feature being checked.
+     * @param featureContext Local context
+     * @return Assigned Variant
+     */
+    public Mono<Variant> getVariantAsync(String feature, Object featureContext) {
+        return generateVariant(feature, featureContext).single();
+    }
+
+    private Mono<Boolean> checkFeature(String feature, Object featureContext) throws FilterNotFoundException {
         if (featureManagementConfigurations.getFeatureManagement() == null
             || featureManagementConfigurations.getOnOff() == null) {
-            return false;
+            return Mono.just(false);
         }
 
         Boolean boolFeature = featureManagementConfigurations.getOnOff().get(feature);
 
         if (boolFeature != null) {
-            return boolFeature;
+            return Mono.just(boolFeature);
         }
 
         Feature featureItem = featureManagementConfigurations.getFeatureManagement().get(feature);
 
         if (featureItem == null || !featureItem.getEvaluate()) {
-            return false;
+            return Mono.just(false);
+        }
+
+        List<Mono<Boolean>> results = new ArrayList<>();
+
+        for (FeatureFilterEvaluationContext featureFilter : featureItem.getEnabledFor().values()) {
+            if (StringUtils.hasText(featureFilter.getName())) {
+                results.add(isFeatureOn(featureFilter, feature, featureContext));
+            }
         }
 
-        Stream<FeatureFilterEvaluationContext> filters = featureItem.getEnabledFor().values().stream()
-            .filter(Objects::nonNull).filter(featureFilter -> featureFilter.getName() != null);
+        if (results.size() == 0) {
+            return Mono.just(false);
+        }
 
         // All Filters must be true
         if (featureItem.getRequirementType().equals("All")) {
-            return filters.allMatch(featureFilter -> isFeatureOn(featureFilter, feature));
+            return Flux.merge(results).reduce((a, b) -> a && b).single();
         }
 
         // Any Filter must be true
-        return filters.anyMatch(featureFilter -> isFeatureOn(featureFilter, feature));
+        return Flux.merge(results).reduce((a, b) -> a || b).single();
```

Comment:
This can be extracted to a method, I am seeing the similar below:

```java

        // All Filters must be true
        if (ALL_REQUIREMENT_TYPE.equals(feature.getRequirementType())) {
            isEnabled = Flux.merge(results).reduce((a, b) -> a && b).single();
        } else {
            // Any Filter must be true
            isEnabled = Flux.merge(results).reduce((a, b) -> a || b).single();
        }
```